src/benchmarks/micro/libraries/System.Text.RegularExpressions/Perf.Regex.Industry.cs

```diff
@@ -0,0 +1,302 @@
+    [BenchmarkCategory(Categories.Libraries)]
+    [SkipTooManyTestCasesValidator]
+    public class Perf_Regex_Industry_SliceSlice
+    {
+        [Params(
+            RegexOptions.None,
+            RegexOptions.IgnoreCase,
+            RegexOptions.Compiled,
+            RegexOptions.Compiled | RegexOptions.IgnoreCase
+#if NET7_0_OR_GREATER
+            , RegexOptions.NonBacktracking
+            , RegexOptions.NonBacktracking | RegexOptions.IgnoreCase
+#endif
+            )]
+        public RegexOptions Options { get; set; }
+
+        private static readonly Regex s_extractWords = new Regex(@"\b(\w+)\b", RegexOptions.Compiled);
+        private Regex[] _regexes;
+        private string _input;
+
+        [GlobalSetup(Target = nameof(Count))]
+        public void Setup()
+        {
+            _input = Perf_Regex_Industry.ReadInputFile("sherlock.txt.gz");
+            _regexes = s_extractWords.Matches(_input).Cast<Match>().Select(m => m.Captures[0].Value).Distinct().Select(w => new Regex(w, Options)).ToArray();
+        }
+
+        [Benchmark]
+        [MinIterationCount(3)] // each iteration takes several seconds
+        public int Count()
+        {
+            int found = 0;
+            foreach (Regex r in _regexes)
+            {
+                found += Perf_Regex_Industry.Count(r, _input);
+            }
+            return found;
+        }
+    }
+
+    /// <summary>Performance tests adapted from https://github.com/rust-lang/regex</summary>
+    [BenchmarkCategory(Categories.Libraries)]
+    [SkipTooManyTestCasesValidator]
+    public class Perf_Regex_Industry_RustLang_Sherlock
+    {
+        [Params(
+            @"Sherlock",
+            @"Holmes",
+            @"Sherlock Holmes",
+            @"(?i)Sherlock",
+            @"(?i)Holmes",
+            @"(?i)Sherlock Holmes",
+            @"Sherlock\s+Holmes",
+            @"Sherlock|Street",
+            @"Sherlock|Holmes",
+            @"Sherlock|Holmes|Watson|Irene|Adler|John|Baker",
+            @"(?i)Sherlock|Holmes|Watson|Irene|Adler|John|Baker",
+            @"Sher[a-z]+|Hol[a-z]+",
+            @"(?i)Sher[a-z]+|Hol[a-z]+",
+            @"Sherlock|Holmes|Watson",
+            @"(?i)Sherlock|Holmes|Watson",
+            @"zqj",
+            @"aqj",
+            @"aei",
+            @"the",
+            @"The",
+            @"(?i)the",
+            @"the\s+\w+",
+            @".*",
+            @"[^\n]*",
+            @"(?s).*",
+            @"\p{L}",
+            @"\p{Lu}",
+            @"\p{Ll}",
+            @"\w+",
+            @"\w+\s+Holmes",
+            @"\w+\s+Holmes\s+\w+",
+            @"Holmes.{0,25}Watson|Watson.{0,25}Holmes",
+            //@"Holmes(?:\s*.+\s*){0,10}Watson|Watson(?:\s*.+\s*){0,10}Holmes", // Too slow with backtracking engine
+            //@"[""'][^""']{0,30}[?!.][""']", // Breaks benchmarkdotnet 13.1
```

Comment:
@adamsitnik, just wanted to draw your attention to this one: it causes benchmark.net to fail to produce any results.

Reply:
@stephentoub I was able to reproduce the BDN bug, will try to send a fix today

Reply:
Fix: https://github.com/dotnet/BenchmarkDotNet/pull/1841

Reply:
That was fast :) Thanks.